cheetah/latticejson.py

```diff
@@ -131,64 +131,103 @@ def encode(self, obj, level=0):
             return json.dumps(obj)
 
 
-def nontorch2feature(value: Any) -> Any:
+def nontorch2feature(
+    value: Any,
+    device: Optional[torch.device] = None,
+    dtype: Optional[torch.dtype] = None,
+) -> Any:
     """
     Convert a value like a `float`, `int`, etc. to a `torch.Tensor` if necessary.
     Values of type `str` and `bool` are not converted, because all currently existing
     `cheetah.Element` subclasses expect these values to not be of type `torch.Tensor`.
 
     :param value: Value to convert to a `torch.Tensor` if necessary.
+    :param device: Device to place the lattice elements on.
+    :param dtype: Data type to use for the lattice elements.
     :return: Value converted to a `torch.Tensor` if necessary.
     """
-    return value if isinstance(value, (str, bool)) else torch.tensor(value)
+    return (
+        value
+        if isinstance(value, (str, bool))
```

Comment:
```suggestion
        if isinstance(value, (str, bool, int))
```
@Hespe ... this feels like the change that should fix the resolution issue, because no tensor value in Cheetah should ever be an int ... but I tested it and it doesn't fix the issue.

This raises the question: Does JSON distinguish between int and float?

**NOTE:** If we make this change, the docstring needs to be changed accordingly.